### PR TITLE
profiles, fti, keycloak, support for dc private repos and general fixes

### PR DESCRIPTION
**Pull request by polographer**
The biggest change on this PR is about profiles, this change means that only enroll is started by default, to load other services you need to specify in the command line with one to start, see the readme for more information

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -45,8 +47,33 @@ Be sure all repositories are up to date with their respective release branches b
...
 
+## Profiles
+By default, the docker-compose is configured to only launch enroll and the required services to run it, however, other profiles can be used to launch other services, the profiles are:
+
+- all: launches all services
+- polypress: launches polypress
+- medicaid_gateway: launches medicaid_gateway / mitc
+- keycloak: launches keycloak
+- fdsh_gateway: launches fdsh_gateway
+- fti: launches fti
```
Comment (mdkaraman) [inline README.md:58]: If this list is meant to be exhaustive, then we should add:
```
enroll_yard_doc: launches enroll documentation site
```

File: README.md [hunk 2]
```diff
@@ -61,6 +88,9 @@ Local aca_entities: it is already mounted and can be changed on the Gemfile, the
 - stimulus reflex on MG
 Docker-compose will patch MG stimulus reflex initializer to ignore dev:cache not being enabled, this is done via a volume mount on the docker-compose.yml file, the file is located at `hotpatches/stimulus_reflex.rb` and is mounted on the container at `/{APP}/config/initializers/stimulus_reflex.rb`. 
 
+- wicked pdf
+Wicked pdf is patched to use the "local" binary (not the one from the wkhtmltopdf-binary-edge gem), this is done via a volume mount on the docker-compose.yml file, the file is located at `hotpatches/wicked_pdf.rb` and is mounted on the container at `/{APP}/config/initializers/wicked_pdf.rb`. The wicked pdf binary is isntalled on the container at `/usr/local/bin/wkhtmltopdf`
```
Comment (mdkaraman) [inline README.md:92]: Just noting a small typo! "isntalled" should be "installed".

File: docker-compose.yml [hunk 3]
```diff
@@ -171,41 +223,47 @@ services:
...
-      dockerfile: ../ea_enterprise/site/medicaid_gateway_service/Dockerfile
-    image: medicaid_gateway:dev-0.1.0
-    volumes:
-      - ../medicaid_gateway:/medicaid_gateway
-      - mcg_rails_cache:/medicaid_gateway/tmp/cache
-      - mcg_bundle:/bundle
-      - mcg_node_modules:/medicaid_gateway/node_modules
-      - mcg_packs:/medicaid_gateway/public/packs
-      - ${PWD}/hotpatches/MG-stimulus.rb:/medicaid_gateway/config/initializers/stimulus_reflex.rb
+  # medicaid_gateway_helper:
```
Comment (mdkaraman) [inline docker-compose.yml:226]: Is there a reason we're commenting this out?  How will we handle client toggling in MG?

**Review by mdkaraman — DISMISSED** [review]
Awesome enhancements!  I've been using this branch for a while now, will be great to see it merged into trunk.  I left a couple comments but nothing that is a major blocker.  We can always touch up the documentation as necessary after this gets merged.

File: site/enroll_service/Dockerfile [hunk 4]
```diff
@@ -42,7 +52,11 @@ RUN chmod +x /usr/local/bin/hackedchromedriver
 RUN gem install bundler -v 2.3.24
 
 COPY . $HOME
+
+ARG BUNDLE_GITHUB__COM
```
Comment (brianhenryhf) [inline site/enroll_service/Dockerfile:56]: with https://github.com/ideacrew/ea_enterprise/pull/24/files#diff-e45e45baeda1c1e73482975a664062aa56f20c03dd9d64a827aba57775bed0d3R148, I don't think this and next line are needed (has been working locally without, and test change confirms). I'd assume same for all the following instances of this, if same `environment` item is present in compose file.

File: docker-compose.yml [hunk 5]
```diff
@@ -118,13 +163,21 @@ services:
...
-      - env.dev
+      - path: ./site/.env.dev
+        required: true
+      - path: ./site/rabbitmq/rabbitmq.dev.env
+        required: true
+      - path: env.dev
+        required: true
+      - path: sso.dev.env
+        required: true
+      - path: env.local
```
Comment (brianhenryhf) [inline docker-compose.yml:176]: I may have missed the idea of `env.dev` initially - if that is gitignored, `env.local` may just be a point of point of potential confusion. Maybe we yank that? (though re-ordering `env.dev` to _after_ `sso.dev.env` would be useful - I use it for changing `LOGOUT_URL` and `LOGIN_URL` so they work in simple running).

**Review by brianhenryhf — APPROVED** [review]
Couple of notes, but no deal-breakers. 👍

File: docker-compose.yml (hunk 5 above)
Comment (polographer) [reply docker-compose.yml:176]: 100% agree, I think we need to re-structure all the env files, initially I wanted to have a "pre-set" but now we have env files that evaluate at different time setting different scopes

However I think we should do it on a different PR, do you mind if we tackle this next ?

File: site/enroll_service/Dockerfile (hunk 4 above)
Comment (polographer) [reply site/enroll_service/Dockerfile:56]: I tried but it didn't work, the build fails

Comment (brianhenryhf) [reply site/enroll_service/Dockerfile:56]: You're totally right. It's been a bit, but I assume I must have been conflating build-time and run-time effects.

**Review by brianhenryhf — APPROVED** [review]
👍

**Pull request merged**